save-demo/src/main/kotlin/com/saveourtool/save/demo/storage/ToolStorage.kt

```diff
@@ -18,9 +19,15 @@ class ToolStorage(
 ) : AbstractFileBasedStorage<ToolKey>(Path.of(configProperties.fileStorage.location) / TOOLS_PATH) {
     @Suppress("DestructuringDeclarationWithTooManyEntries")
     override fun buildKey(rootDir: Path, pathToContent: Path): ToolKey {
-        val (_, ownerName, toolName, vcsTagName, executableName) = pathToContent.pathNamesTill(rootDir)
+        val (ownerName, toolName, vcsTagName, executableName) = pathToContent.pathNamesTill(rootDir)
         return ToolKey(ownerName, toolName, vcsTagName, executableName)
     }
 
     override fun buildPathToContent(rootDir: Path, key: ToolKey): Path = rootDir / key.ownerName / key.toolName / key.vcsTagName / key.executableName
+
+    override fun isKey(rootDir: Path, pathToContent: Path): Boolean = pathToContent.name != DS_STORE
```

Comment:
suggestion to path `pathPartsCount` in constructor instead of overriding `isKey` for mac specific use case

Reply:
Oh sure

Reply:
And what if `.DS_Store` is in the directory with the tool?

Reply:
Fixed it

Reply:
got you, discussed offline --- agreed to add this fix in `AbstractFileBasedStorage`